### PR TITLE
Update replication docs to clarify permissions

### PR DESCRIPTION
**Pull request by poornas**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

File: docs/bucket/replication/README.md [hunk 1]
```diff
@@ -18,8 +18,26 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
 Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
 ```
 
-Note that the admin needs *s3:GetReplicationConfigurationAction* permission on source cluster. The credential used at the destination requires *s3:ReplicateObject* permission. Once successfully created and authorized this generates a replication target ARN.  The command below lists all the currently authorized replication targets:
+Note that the admin user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on source cluster if admin user is not the default minio admin. The access credentials used at the destination requires *s3:ReplicateObject* permission.
```
Comment (harshavardhana) [inline docs/bucket/replication/README.md]:
```suggestion
> NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.
```

File: docs/bucket/replication/README.md [hunk 2]
```diff
@@ -18,8 +18,26 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
...
+```
+    "s3:ReplicateTags",
+    "s3:GetObject",
+    "s3:GetObjectVersion",
+    "s3:GetObjectVersionTagging",
+    "s3:PutObject",
+    "s3:ReplicateObject"
+    "s3:GetBucketVersioning"
+```
+Once successfully created and authorized the `mc admin bucket remote add` command generates a replication target ARN.  The command below lists all the currently authorized replication targets:
```
Comment (harshavardhana) [inline docs/bucket/replication/README.md]: It would be worth to provide an entire JSON document with full examples @poornas

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10536-7fc6cb7

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#12fa8aa6-2239-4f2c-b94e-f55252701140) |
<a name="12fa8aa6-2239-4f2c-b94e-f55252701140"></a>
### 10536-c840413/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31681
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ff788cc348d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 1 minutes and 3 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5f6942178f5e0"
  },
  "duration": "40635",
  "status": "FAIL",
  "error": "Error executing \"CreateBucket\" on \"http://minio-dev8.minio.io:31681/aws-sdk-php-5f6942178f5e0\"; AWS HTTP error: Client error: `PUT http://minio-dev8.minio.io:31681/aws-sdk-php-5f6942178f5e0` resulted in a `404 Not Found` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Mes (truncated...)\n NoSuchBucket (client): The specified bucket does not exist - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><BucketName>aws-sdk-php-5f6942178f5e0</BucketName><Resource>/aws-sdk-php-5f6942178f5e0</Resource><RequestId>1636F2E4C748E873</RequestId><HostId>faac76fe-aeab-4be3-855a-90044632e4a8</HostId></Error>"
}
(3/15) Running aws-sdk-ruby tests ... done in 23 seconds
(4/15) Running awscli tests ... done in 2 minutes and 53 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 5 minutes and 58 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 41 seconds
(8/15) Running minio-go tests ... done in 6 minutes and 28 seconds
(9/15) Running minio-java tests ... done in 10 minutes and 44 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 37 seconds
(11/15) Running minio-py tests ... FAILED in 17 minutes and 39 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea",
    "object_name": "c96709ef-960c-41ee-8c17-6b619fdaa6ac",
    "presigned_get_object_url": "http://minio-dev8.minio.io:31681/minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea/c96709ef-960c-41ee-8c17-6b619fdaa6ac?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioazure%2F20200922%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200922T010445Z&X-Amz-Expires=120&X-Amz-SignedHeaders=host&X-Amz-Signature=c112a76c0c7a40f53bc45be3406d247e83617da4931e6aa8e8e5e8eb66363e45",
    "response.status": 500,
    "response.reason": "Internal Server Error",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"473\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"etag\": [\"ETag\", \"\\\"6d2040290066a0bde02c04e3a24c34d6\\\"\"], \"last-modified\": [\"Last-Modified\", \"Tue, 22 Sep 2020 01:04:45 GMT\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2020-09-22T00-05-14Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"1636F588D059E23E\"], \"x-cache\": [\"X-Cache\", \"HIT\"], \"x-cache-lookup\": [\"X-Cache-Lookup\", \"HIT\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 22 Sep 2020 01:04:45 GMT\"]}}",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InternalError</Code><Message>We encountered an internal error, please try again.</Message><Key>c96709ef-960c-41ee-8c17-6b619fdaa6ac</Key><BucketName>minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea</BucketName><Resource>/minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea/c96709ef-960c-41ee-8c17-6b619fdaa6ac</Resource><RequestId>1636F588D059E23E</RequestId><HostId>faac76fe-aeab-4be3-855a-90044632e4a8</HostId></Error>"
  },
  "message": "InternalError: message: We encountered an internal error. Please try again.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1349, in test_presigned_get_object_expiry\n    object_name).get_exception()\nminio.error.InternalError: InternalError: message: We encountered an internal error. Please try again.\n",
  "duration": 1098
}
(11/15) Running s3cmd tests ... done in 2 minutes and 22 seconds
(12/15) Running s3select tests ... done in 1 minutes and 7 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10536-c840413

**Review by harshavardhana — APPROVED** [review]

File: docs/bucket/replication/README.md [hunk 3]
```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
...
+  },
+  {
+   "Effect": "Allow",
+   "Action": [
+    "s3:ReplicateTags",
+    "s3:GetObject",
+    "s3:GetObjectVersion",
+    "s3:GetObjectVersionTagging",
+    "s3:PutObject",
+    "s3:ReplicateObject"
```
Comment (kannappanr) [inline docs/bucket/replication/README.md:64]: Do we need these permissions?

"s3:PutObjectRetention",
"s3:PutObjectLegalHold",
"s3:PutObjectTagging",

Comment (poornas) [reply docs/bucket/replication/README.md:64]: only if object locking is enabled.

**Review by kannappanr — APPROVED** [review]
LGTM

File: docs/bucket/replication/README.md [hunk 4]
```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
 Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
 ```
 
-Note that the admin needs *s3:GetReplicationConfigurationAction* permission on source cluster. The credential used at the destination requires *s3:ReplicateObject* permission. Once successfully created and authorized this generates a replication target ARN.  The command below lists all the currently authorized replication targets:
+> NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.
```
Comment (klauspost) [inline docs/bucket/replication/README.md]:
```suggestion
>  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster. We do not recommend running root credentials/super admin with replication, instead create a dedicated user. The access credentials used at the destination requires *s3:ReplicateObject* permission.
```

File: docs/bucket/replication/README.md [hunk 5]
```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
...
 
+Minimal policy on source bucket should have following permissions on the source bucket.
```
Comment (klauspost) [inline docs/bucket/replication/README.md]:
```suggestion
The *source* bucket should have following minimal permission policy:
```

File: docs/bucket/replication/README.md [hunk 6]
```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
...
+    "s3:GetBucketLocation",
+    "s3:GetBucketVersioning"
+   ],
+   "Resource": [
+    "arn:aws:s3:::srcbucket"
+   ]
+  }
+}
+```
+The access key configured to perform replication on the target cluster should have minimal permissions as in example below.
```
Comment (klauspost) [inline docs/bucket/replication/README.md]:
```suggestion
The access key provided for the replication *target* cluster should have these minimal permissions:
```

File: docs/bucket/replication/README.md [hunk 7]
```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
...
+   ],
+   "Resource": [
+    "arn:aws:s3:::destbucket/*"
+   ]
+  }
+ ]
+}
+
+```
+Once successfully created and authorized the `mc admin bucket remote add` command generates a replication target ARN.  The command below lists all the currently authorized replication targets:
```
Comment (klauspost) [inline docs/bucket/replication/README.md]:
```suggestion
Once successfully created and authorized, the `mc admin bucket remote add` command generates a replication target ARN.  This command lists all the currently authorized replication targets:
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#6c0fe684-b65f-4844-abae-0b53f04cdf20) |
<a name="6c0fe684-b65f-4844-abae-0b53f04cdf20"></a>
### 10536-cbdd952/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31289
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d6dbe33ed92d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 9 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 3 minutes and 2 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 55 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 40 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 57 seconds
{
  "args": {
    "bucketName": "minio-go-test-swcyzv0094kcw20y",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1045,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 10 minutes and 54 seconds
(10/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 82,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:682:10) at endReadableNT (_stream_readable.js:1252:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(10/15) Running minio-py tests ... done in 18 minutes and 40 seconds
(11/15) Running s3cmd tests ... done in 2 minutes and 19 seconds
(12/15) Running s3select tests ... done in 40 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10536-cbdd952

**Pull request merged**